packages/opentelemetry-api/src/trace/instrumentation/Plugin.ts

```diff
@@ -37,12 +38,14 @@ export interface Plugin<T = any> {
    * @param moduleExports The value of the `module.exports` property that would
    *     normally be exposed by the required module. ex: `http`, `https` etc.
    * @param TracerProvider a tracer provider.
+   * @param MeterProvider a meter provider.
    * @param logger a logger instance.
    * @param [config] an object to configure the plugin.
    */
   enable(
     moduleExports: T,
     TracerProvider: TracerProvider,
+    MeterProvider: MeterProvider,
```

Comment:
My biggest concern here is to have the backward compatibility not to change anything.
What we could do is this:
```
export interface BothProviders {
  tracer: TracerProvider;
  meter?: MeterProvider;
}
export type Providers = BothProviders | TracerProvider;

enable(
moduleExports: T,
TracerProvider: Providers
...
)

// and then

export abstract class BaseAbstractPlugin<T> implements Plugin<T> {
  abstract enable(
    moduleExports: T,
    tracerProvider: Providers,
    logger: Logger,
    config?: PluginConfig
  ): T;
...

// and

export abstract class BasePlugin<T> extends BaseAbstractPlugin<T>
  implements Plugin<T> {
  enable(
    moduleExports: T,
    providers: Providers,
    logger: Logger,
    config?: PluginConfig
  ): T {
    this._moduleExports = moduleExports;
    let tracerProvider: TracerProvider;
    let meterProvider: MeterProvider;
    if ((providers as BothProviders).tracer) {
      tracerProvider = (providers as BothProviders).tracer;
      meterProvider = (providers as BothProviders).meter;
    } else {
      tracerProvider = providers as TracerProvider;
    }

```

This way we won't have problem with plugins, keep the backward compatibility and once we upgrade contrib repo etc. we can get rid of 
```
export type Providers = BothProviders | TracerProvider;
```
and just leave `BothProviders`

Of course naming `BothProviders` should be something different :)

Reply:
The other way would be to add it at the end of the argument list, which might be simpler.

Reply:
Is there any reason that i miss on why we should pass the providers as arguments ? (the alternative being using the ones registered in `@opentelemetry/api`)

Reply:
> Is there any reason that i miss on why we should pass the providers as arguments ? (the alternative being using the ones registered in `@opentelemetry/api`)

I'm curious to know what others think about this. But based on what @obecny said above, it seems like we have to keep in backward compatible. Also previously you could enable plugins in the config of a tracer provider, so you didn't need to register that tracer provider in order to use plugins. I personally think using the registered providers might be a bit more confusing than passing providers to the `PluginManager`

Reply:
>Is there any reason that i miss on why we should pass the providers as arguments ? (the alternative being using the ones registered in @opentelemetry/api)

As far as I'm aware, this is to support the use-case where a user has multiple tracer providers in their application.

Reply:
> As far as I'm aware, this is to support the use-case where a user has multiple tracer providers in their application.

Well the API only allow for one anyway so the user will end up registering only the one he want the plugins to use ? Others will be created manually and kept out of the api.

It seems that both [Go](https://github.com/open-telemetry/opentelemetry-go-contrib/blob/master/instrumentation/github.com/gorilla/mux/mux.go#L131), [Java](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/d6e39f89e6dd2a9f15c173a076b4302960c83b2c/agent-bootstrap/src/main/java/io/opentelemetry/auto/bootstrap/instrumentation/decorator/DatabaseClientTracer.java#L47) and [Ruby](https://github.com/open-telemetry/opentelemetry-ruby/blob/47929de5b32a76d48febbcc341deb6eb8101fca3/api/lib/opentelemetry/instrumentation/base.rb#L185) does the same.

Reply:
If we're going to do that, we should do a clean sweep and remove the injection from all of the plugins.

Reply:
I think the first step would be to *not* add an argument for the `meterProvider` and then make a follow-up PR to remove the `tracerProvider`, WDYT ? I volonteer to do the later

Reply:
Let's talk about this in the SIG today? would like other @open-telemetry/javascript-maintainers opinions on changing the plugin interface to be backwards incompatible.